### PR TITLE
Watsonx documentation update

### PR DESCRIPTION
**Pull request by mdswift**
- Add links to watsonx use cases: complex tasks, conversational search, summarization
- Add details on how to find the watsonx project id

### CODE REVIEW

File: integrations/extensions/starter-kits/language-model-watsonx/README.md
Comment (jwm4) [inline integrations/extensions/starter-kits/language-model-watsonx/README.md:24]: Why is this ◊ here?  Is it a typo?

File: integrations/extensions/starter-kits/language-model-watsonx/README.md
Comment (jwm4) [inline integrations/extensions/starter-kits/language-model-watsonx/README.md:7]: The watsonx kit for this one is not merged yet.  Should we wait to merge this change until that one is merged too?

**Review by jwm4 — CHANGES_REQUESTED** [review]

File: integrations/extensions/starter-kits/language-model-watsonx/README.md
Comment (mdswift) [reply integrations/extensions/starter-kits/language-model-watsonx/README.md:7]: Yes, it makes sense to wait.

File: integrations/extensions/starter-kits/language-model-watsonx/README.md
Comment (mdswift) [reply integrations/extensions/starter-kits/language-model-watsonx/README.md]: Yes, a typo. Fixed.

**Review by jwm4 — APPROVED** [review]
This was blocked on the watsonx complex info kit being merged.  Now that one is merged, so it is time to merge these documentation updates too.

**Pull request merged**
